elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -206,6 +249,67 @@ public void commit() {
     }
 
     @Override
+    public <T> Collection filterSortOrPaginateCollection(final Collection collection, final Class<T> entityClass,
+                                                         final EntityDictionary dictionary,
+                                                         final Optional<Set<Predicate>> filters,
+                                                         final Optional<Sorting> sorting,
+                                                         final Optional<Pagination> pagination) {
+        if (((collection instanceof AbstractPersistentCollection))
+                && (filters.isPresent() || sorting.isPresent() || pagination.isPresent())) {
+
+            String filterString = "";
+
+            // apply filtering - eg where clause's
+            if (filters.isPresent()) {
+                filterString += hqlFilterOperation.applyAll(filters.get());
+            }
+
+            // add sorting into HQL string query generation
+            if (sorting.isPresent() && !sorting.get().isEmpty()) {
+
+                final Map<String, Sorting.SortOrder> validSortingRules = sorting.get().getValidSortingRules(
+                        entityClass, dictionary
+                );
+                String additionalHQL = "";
+                if (!validSortingRules.isEmpty()) {
+
+                    final List<String> ordering = new ArrayList<>();
+                    // pass over the sorting rules
+                    validSortingRules.entrySet().stream().forEachOrdered(entry ->
+                            ordering.add(entry.getKey() + " " + (entry.getValue().equals(Sorting.SortOrder.desc)
+                                    ? "desc"
+                                    : "asc"))
+                    );
+                    additionalHQL += "order by " + StringUtils.join(ordering, ",");
+                }
+                if (!additionalHQL.isEmpty()) {
+                    filterString += additionalHQL;
+                }
+            }
+
+            if (filterString.length() != 0) {
+                Query query = session.createFilter(collection, filterString);
```

Comment:
regardless of actual filters (eg. select 'fields' from Table WHERE x in y, etc) we can still support ordering if nothing else here, so filter keyword is a bit ambiguous since we have Set<Predicate> filters, and then Sorting which is also a filter, but separate from HQL filters and the notion of Criterion filters, so just want to ensure things here make sense.